app/api/user.py

```diff
@@ -43,3 +43,101 @@ def user_assign_role(user_id, role):
         return jsonify(ret), 200
     else:
         return jsonify(msg="invalid role"), 500
+
+    
+    
+   
+@bp.route("/recent_activity", methods=["GET"])
+@jwt_required
+def recent_activity():
+        # First take date time where weekly report is to be dealt with
+        today = datetime.date.today()
+        last_sunday = today - datetime.timedelta(days=(today.weekday() + 1))
+        last_monday = today - datetime.timedelta(days=(today.weekday() + 8))
+        
+        # Get current user 
+        current_user = get_current_user()
+        # Check if current user is Admin or manager
+        if current_user["role"] == "Admin" or "manager":
+            # First find the users which have the manager id in it's document
+            users = mongo.db.users.find({
+                "managers": {
+                    "$elemMatch": {"_id": str(current_user["_id"])}
+                }
+                })
+            # Make a list of User_id and then fetch all the users/juniors belong to that manager
+            user_ids = []
+            for user in users:
+                user_ids.append(str(user['_id']))
+            # Now find weekly reports of all those users_id in the above list whose report is not reviewd   
+            for data in  user_ids:
+                docs = mongo.db.reports.find({
+                    "user":str(data),
+                    "review": {'$exists': False},
+                    "type" : "weekly",
+                    "created_at": {
+                    "$gte": datetime.datetime(last_monday.year, last_monday.month, last_monday.day),
+                    "$lte": datetime.datetime(last_sunday.year, last_sunday.month, last_sunday.day)
+                     }  
+                })
+                docs = [serialize_doc(doc) for doc in docs]
+                # Append those user records whose not been reviewd in user_id list
+                user_id = []
+                for user in docs:
+                    user_id.append(user)
+                # Then if we find that data exist in user_id report than update the manager recent activity with a date and message    
+                if user_id is not None: 
+                    ret = mongo.db.recent_activity.update({
+                    "user": str(current_user["_id"]),},
+                    {"$set":{"recent_activity":{
+                    "created_at": datetime.datetime.now(),
+                    "Message":"Your have not reviewed your juniors weekly report"
+                    }}},upsert=True)

```

Comment:
a manager himself has to do daily reports himself as manager also works. on project. so if he missed checkin daily or weekly. he also needs to get a that msg